### DISCUSSION TITLE
play-scala-slick how to get database password from outside the config file

### DISCUSSION THREAD

**Discussion by Tasselmi** [playframework/playframework#11914] [General]
`play-scala-slick` how to get database password from outside the config file？for example，from Apollo configuration center.
I don't know how to get database password or other dynamic configurations from other than `.conf` file. 
please teach me. thanks.

**Comment by tsuyoshizawa** [comment]
You can read in environment variables or `include` another config file to overwrite the configuration values.
https://www.playframework.com/documentation/2.8.x/ConfigFile

**Comment by Tasselmi** [comment]
> example

I mean from outside the config file， for example from http response. We have all of our passwords stored in our configuration center. We have to fetch the password from http request and apply the result when connecting DB with slick.

**Comment by irumiha** [comment]
In your local config file you can do `include url("http://example.com")`. The config library will issue a GET to the specified endpoint. The response should be syntactically valid HOCON file. If Apollo can't give that out the box you could write a small tool that listens to such requests and then issues the appropriate API calls to Apollo to prepare the valid response.

**Reply by Tasselmi** [reply]
Thank you so much
